draft-yasskin-http-origin-signed-responses.md

```diff
@@ -200,8 +206,70 @@ present parameters MUST have the following values:
 : An unsigned integer (Section 4.1 of {{!I-D.ietf-httpbis-header-structure}})
   representing a Unix time.
 
-The "certUrl" and "validityUrl" parameters are *not* signed, so intermediates can
-update them with pointers to cached versions.
+The "certUrl" parameter is *not* signed, so intermediates can update it with a
+pointer to a cached version.
+
+### Examples ### {#example-signature-header}
+
+The following header is included in the response for an exchange with effective
+request URI `https://example.com/resource.html`. Newlines are added for
+readability.
+
+~~~http
+Signature:
+ sig1;
+  sig=*MEUCIQDXlI2gN3RNBlgFiuRNFpZXcDIaUpX6HIEwcZEc0cZYLAIga9DsVOMM+g5YpwEBdGW3sS+bvnmAJJiSMwhuBdqp5UY;
+  integrity="mi";
+  validityUrl="https://example.com/resource.validity.1511128380";
+  certUrl="https://example.com/oldcerts";
+  certSha256=*W7uB969dFW3Mb5ZefPS9Tq5ZbH5iSmOILpjv2qEArmI;
+  date=1511128380; expires=1511733180,
+ sig2;
+  sig=*MEQCIGjZRqTRf9iKNkGFyzRMTFgwf/BrY2ZNIP/dykhUV0aYAiBTXg+8wujoT4n/W+cNgb7pGqQvIUGYZ8u8HZJ5YH26Qg;
+  integrity="mi";
+  validityUrl="https://example.com/resource.validity.1511128380";
+  certUrl="https://example.com/newcerts";
+  certSha256=*J/lEm9kNRODdCmINbvitpvdYKNQ+YgBj99DlYp4fEXw;
+  date=1511128380; expires=1511733180,
+ srisig;
+  sig=*lGZVaJJM5f2oGczFlLmBdKTDL+QADza4BgeO494ggACYJOvrof6uh5OJCcwKrk7DK+LBch0jssDYPp5CLc1SDA
+  integrity="mi";
+  validityUrl="https://example.com/resource.validity.1511128380";
+  ed25519Key=*zsSevyFsxyZHiUluVBDd4eypdRLTqyWRVOJuuKUz+A8
+  date=1511128380; expires=1511733180,
+ thirdpartysig;
+  sig=*MEYCIQCNxJzn6Rh2fNxsobktir8TkiaJYQFhWTuWI1i4PewQaQIhAMs2TVjc4rTshDtXbgQEOwgj2mRXALhfXPztXgPupii+;
+  integrity="mi";
+  validityUrl="https://thirdparty.example.com/resource.validity.1511161860";
+  certUrl="https://thirdparty.example.com/certs";
+  certSha256=*UeOwUPkvxlGRTyvHcsMUN0A2oNsZbU8EUvg8A9ZAnNc;
+  date=1511133060; expires=1511478660,
+~~~
+
+There are 4 signatures: 2 from different secp256r1 certificates within
+`https://example.com/`, one using a raw ed25519 public key that's also
+controlled by `example.com`, and a fourth using a secp256r1 cross-signing
+certificate owned by `thirdparty.example.com`.
```

Comment:
Could you explain what you mean by 'cross-signing certificate' here?

That is, 'normally' cross-signing refers to the same subject entity being validated by two different issuers - e.g., a given intermediate "Intermediate Foo" can be signed by both "CA A" and "CA B".

This makes me think that the use case here is `thirdparty.example.com` asserting that the content is truly from `example.com`, rather than being `example.com` proving that the `thirdpartysig` covers the resources from `thirdparty.example.com`.

If it's the latter (`thirdparty.example.com` certifying the content on `example.com`), then it seems we're introducing yet another Trusted Third Party into Web Packaging, beyond the CA model. That seems a radical departure from the TLS guarantees, and perhaps a subtle aspect I failed to realize - that this would allow non-CA entities to certify certificates.

Reply:
I mean for all four of these signatures to sign the resource, not any other certificates. (Remember, this is currently about a single exchange, not for a bundle of exchanges that might contain resources from several origins.) One use case might be for `thirdparty.example.com` to be a binary transparency log whose signature certifies that the resource is contained in the log. I'm adding a more explicit statement of the cross-signing use cases in https://github.com/WICG/webpackage/pull/112/files#diff-a2197f2cbc199dc1463dc10213fa40d9R862.

I'm going to remove the use of "cross-signing" lest it imply signing certificates to anyone else, and then merge so that I can publish a new I-D.